pkg/mutation/path/parser/node.go

```diff
@@ -32,28 +32,70 @@ type Path struct {
 	Nodes []Node
 }
 
+var _ Node = Path{}
+
 func (r Path) Type() NodeType {
 	return PathNode
 }
 
+func (r Path) DeepCopy() Path {
+	out := Path{
+		Nodes: make([]Node, len(r.Nodes)),
+	}
+	for i := 0; i < len(r.Nodes); i++ {
+		var c Node
+		switch n := r.Nodes[i].(type) {
+		case Path:
+			c = n.DeepCopy()
+		case List:
+			c = n.DeepCopy()
+		case Object:
+			c = n.DeepCopy()
+		}
+		out.Nodes[i] = c
```

Comment:
What should happen in the `default` case? Should we copy it as-is? Should we panic?

Reply:
Good question -- code now runs DeepCopyNode, though, so I think we're safe here, presuming DeepCopy is implemented on the Node type correctly.